kymatio/scattering2d/filter_bank.py

```diff
@@ -36,35 +36,39 @@ def filter_bank(M, N, J, L=8):
     filters = {}
     filters['psi'] = []
 
-    offset_unpad = 0
     for j in range(J):
         for theta in range(L):
             psi = {}
             psi['j'] = j
             psi['theta'] = theta
             psi_signal = morlet_2d(M, N, 0.8 * 2**j,
                 (int(L-L/2-1)-theta) * np.pi / L,
-                3.0 / 4.0 * np.pi /2**j, 4.0/L, offset=offset_unpad)
+                3.0 / 4.0 * np.pi /2**j, 4.0/L)
             psi_signal_fourier = fft2(psi_signal)
+            # drop the imaginary part, it is zero anyway
+            psi_signal_fourier = np.real(psi_signal_fourier)
             for res in range(min(j + 1, J - 1)):
                 psi_signal_fourier_res = periodize_filter_fft(
                     psi_signal_fourier, res)
-                psi[res] = torch.FloatTensor(
-                    np.stack((np.real(psi_signal_fourier_res),
-                    np.imag(psi_signal_fourier_res)), axis=2))
+                # add a trailing singleton dimension to mark it as non-complex
+                psi_signal_fourier_res = psi_signal_fourier_res[..., np.newaxis]
+                psi[res] = torch.FloatTensor(psi_signal_fourier_res)
```

Comment:
the reason numpy is used here is to make the filters backend agnostic. I agree that right now, the software is not ready for a tf integration, however I believe it is a good practice to keep numpy here

Reply:
It created a FloatTensor before and it creates a FloatTensor afterwards. I don't like it, but changing this should be another PR.

Reply:
Yeah I would tend to agree with @f0k. Let's keep things separate.

Reply:
If @edouardoyallon agrees, I suggest we mark this and the other comment as resolved.

Reply:
goahead; i'll open an issue about that if nobody does it!

Reply:
Beat you to it.